vowpalwabbit/global_data.h

```diff
@@ -526,13 +532,14 @@ struct vw
   std::vector<std::string> limit_strings; // descriptor of feature limits
   uint32_t limit[256];//count to limit features by
   uint64_t affix_features[256]; // affixes to generate (up to 16 per namespace - 4 bits per affix)
-  bool     spelling_features[256]; // generate spelling features for which namespace
+  bool spelling_features[256]; // generate spelling features for which namespace
   std::vector<std::string> dictionary_path;  // where to look for dictionaries
   std::vector<feature_dict*> namespace_dictionaries[256]; // each namespace has a list of dictionaries attached to it
   std::vector<dictionary_info> loaded_dictionaries; // which dictionaries have we loaded from a file to memory?
 
   void(*delete_prediction)(void*); bool audit; //should I print lots of debugging information?
   bool quiet;//Should I suppress progress-printing of updates?
+  bool help_requested;
```

Comment:
Semantics?  Why do we need to add this?  (We generally want to remove fields here.)

Reply:
In order to print help, all possible options must have been seen already. We only see the options available after `parse_modules` so we must save the intent as we parse the option and then carry out the action later.